web/server/codechecker_server/database/db_cleanup.py

```diff
@@ -223,13 +172,70 @@ def remove_unused_analysis_info(session_maker):
 
             to_delete = map(lambda x: x[0], to_delete)
 
+            total_count = 0
             for chunk in util.chunks(to_delete, CHUNK_SIZE):
-                session.query(AnalysisInfo) \
+                count = session.query(AnalysisInfo) \
                     .filter(AnalysisInfo.id.in_(chunk)) \
                     .delete(synchronize_session=False)
-                session.commit()
+                if count:
+                    total_count += count
+
+            if total_count:
+                LOG.debug("%d dangling analysis info deleted.", total_count)
+
+            session.commit()
 
             LOG.debug("Garbage collection of dangling analysis info finished.")
         except (sqlalchemy.exc.OperationalError,
                 sqlalchemy.exc.ProgrammingError) as ex:
             LOG.error("Failed to remove dangling analysis info: %s", str(ex))
+
+
+def upgrade_severity_levels(session_maker, checker_labels):
+    """
+    Updates the potentially changed severities at the reports.
+    """
+    LOG.debug("Upgrading severity levels started...")
+
+    with DBSession(session_maker) as session:
+        try:
+            count = 0
+            for analyzer in sorted(checker_labels.get_analyzers()):
+                checkers_for_analyzer_in_database = session \
+                    .query(Checker.id,
+                           Checker.checker_name,
+                           Checker.severity) \
+                    .filter(Checker.analyzer_name == analyzer) \
+                    .all()
+                for checker_row in checkers_for_analyzer_in_database:
+                    checker: str = checker_row.checker_name
+                    old_severity_db: int = checker_row.severity
+                    old_severity: str = \
+                        Severity._VALUES_TO_NAMES[old_severity_db]
+                    new_severity: str = \
+                        checker_labels.severity(checker, analyzer)
```

Comment:
There might be a slight issue here, because `(analyzer, checker)` comes from the database, which means it will be, for example `(cppcheck, unusedVariable)`,  but the config file, unfortunately, contains `cppcheck-unusedVariable` as the _checker name_, resulting in the severity becoming wiped.

I will do two things to fix this:

 1. Of course, fix the logic here, so the "new style" checker names are also attempted from the config file.
 2. Prevent the wiping of severity to `0` in addition to giving an explicit :warning: `[WARNING]` about the fact.